kedro/runner/runner.py

```diff
@@ -294,10 +297,22 @@ def run_node(
             asynchronously with threads. Defaults to False.
         session_id: The session id of the pipeline run.
 
+    Raises:
+        ValueError: Raised if is_async is set to True for nodes wrapping
+        generator functions.
+
     Returns:
         The node argument.
 
     """
+    if is_async and inspect.isgeneratorfunction(node.func):
+        raise ValueError(
+            f"Async data loading and saving does not work with "
+            f"nodes wrapping generator functions. Please make "
+            f"sure you don't use yield anywhere in your function "
+            f"in node {str(node)}"
```

Comment:
Technically node is only a wrapper object to functions, but shortening it might be worth it, since people use node/function interchangeably when talking about Kedro projects.